controller/up.go

```diff
@@ -89,9 +89,9 @@ func compress(src string, buf io.Writer) error {
 	return nil
 }
 
-func (c *Controller) Upload(ctx context.Context, req *entity.UploadRequest) (*entity.UpResponse, error) {
+func (c *Controller) Upload(ctx context.Context, req *entity.UploadRequest, src string) (*entity.UpResponse, error) {
```

Comment:
This req is a Upload Request, with a couple properties on it

What I mean is add a property along the lines of "RootDir" and pass it into the function as such, vs creating a new parameter